students/Hebdowska_Grazyna/lesson_06/cubes.py

```diff
@@ -0,0 +1,42 @@
+def input_value():
+    nm = input()
+    nm = nm.split(" ")
+    n = int(nm[0])
+    m = int(nm[1])
+    alice = []
+    bob = []
+    for i in range(n):
+        alice.append(int(input()))
+    for i in range(m):
+        bob.append(int(input()))
+    cube = {"alice": alice, "bob": bob}
+    return cube
+
+
+def only_a(cube, k1, k2, only):
+    oa = []
+    both = []
+    for i in cube[k1]:
+        if i not in cube[k2]:
+            oa.append(i)
+        else:
+            both.append(i)
+    if only is False:
+        oa = both
+    oa.sort()
+    return oa
+
+
+def result(lis):
+    print(len(lis))
+    for i in lis:
+        print(i)
+
+
+cube = input_value()
+only_alice = only_a(cube, "alice", "bob", True)
+only_bob = only_a(cube, "bob", "alice", True)
+cubes = only_a(cube, "alice", "bob", False)
+result(cubes)
+result(only_alice)
+result(only_bob)
```

Comment:
Your solution works but it might be easier to be implemented by using sets()
Example below:
```
n, m = [int(i) for i in input('Numbers: ').split()]
alice_set = set()
bob_set = set()
for i in range(n):
    alice_set.add(int(input('Alice set number: ')))

for i in range(m):
    bob_set.add(int(input('Bob set number: ')))


intersection = sorted(alice_set.intersection(bob_set))
print(len(intersection))
for i in intersection:
    print(i)

difference = sorted(alice_set.difference(bob_set))
print(len(difference))
for i in difference:
    print(i)

difference_update = sorted(bob_set.difference(alice_set))
print(len(difference_update))
for i in difference_update:
    print(i)
```